streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StreamSourceNode.java

```diff
@@ -71,6 +78,21 @@ public Pattern topicPattern() {
         return consumedInternal.valueSerde();
     }
 
+    // We "merge" source nodes into a single node under the hood if a user tries to read in a source topic multiple times
+    public void merge(final StreamSourceNode<?, ?> other) {
+        final AutoOffsetReset resetPolicy = consumedInternal.offsetResetPolicy();
+        if (resetPolicy != null && !resetPolicy.equals(other.consumedInternal().offsetResetPolicy())) {
```

Comment:
What should happen if this reset policy is `null` and the other is not `null`? I guess we should also throw in that case, don't we?

Reply:
Ack, good catch